src/lib/Canvas.js

```diff
@@ -759,4 +740,50 @@ function makeCharMap() {
     return map;
 }
 
+function getCachedContext(canvasElement, type) {
+    var gc = canvasElement.getContext(type || '2d'),
+        props = {},
+        values = {};
+
+    // Stub out all the prototype members of the canvas 2D graphics context:
+    Object.keys(Object.getPrototypeOf(gc)).forEach(makeStub);
+
+    // Some older browsers (e.g., Chrome 40) did not have all members of canvas
+    // 2D graphics context in the prototype so we make this additional call:
+    Object.keys(gc).forEach(makeStub);
+
+    function makeStub(key) {
+        if (
+            !(key in props) &&
+            !/^(webkit|moz|ms|o)[A-Z]/.test(key) &&
+            typeof gc[key] !== 'function'
+        ) {
+            Object.defineProperty(props, key, {
+                get: function() {
+                    return (values[key] = values[key] || gc[key]);
```

Comment:
BTW: The mediator is not mandatory — but of course the cache would get out of sync on setter calls and getter calls would be way, way slower.